File: generate-icons.py
#!/usr/bin/env python3
"""Generate PWA icons for VacaTrack"""

from PIL import Image, ImageDraw, ImageFont
import os

# Colors from the app
BG_COLOR = (255, 254, 251)  # #fffefb
PRIMARY_COLOR = (0, 102, 140)  # #00668c
ACCENT_COLOR = (113, 196, 239)  # #71c4ef
WHITE = (255, 255, 255)

def generate_icon(size, filepath):
    """Generate a single icon of given size"""
    # Create image with background
    img = Image.new('RGB', (size, size), BG_COLOR)
    draw = ImageDraw.Draw(img)

    # Calculate dimensions
    center_x = size // 2
    center_y = size // 2
    radius = int(size * 0.35)

    # Draw primary circle
    draw.ellipse(
        [center_x - radius, center_y - radius,
         center_x + radius, center_y + radius],
        fill=PRIMARY_COLOR
    )

    # Draw accent circle
    inner_radius = int(radius * 0.65)
    draw.ellipse(
        [center_x - inner_radius, center_y - inner_radius,
         center_x + inner_radius, center_y + inner_radius],
        fill=ACCENT_COLOR
    )

    # Draw "V" text
    font_size = int(size * 0.5)
    try:
        # Try to use a nice system font
        font = ImageFont.truetype("/System/Library/Fonts/Supplemental/Arial Bold.ttf", font_size)
    except:
        try:
            font = ImageFont.truetype("/System/Library/Fonts/Helvetica.ttc", font_size)
        except:
            # Fallback to default font
            font = ImageFont.load_default()

    # Calculate text position
    text = "V"
    bbox = draw.textbbox((0, 0), text, font=font)
    text_width = bbox[2] - bbox[0]
    text_height = bbox[3] - bbox[1]
    text_x = center_x - text_width // 2
    text_y = center_y - text_height // 2 - int(size * 0.05)  # Slight adjustment

    # Draw text
    draw.text((text_x, text_y), text, fill=WHITE, font=font)

    # Save
    img.save(filepath, 'PNG')
    print(f"Generated {filepath}")

def main():
    """Generate all required icons"""
    print("Generating VacaTrack PWA icons...\n")

    # Create public directory if it doesn't exist
    os.makedirs('public', exist_ok=True)

    # Generate icons
    generate_icon(192, 'public/pwa-192x192.png')
    generate_icon(512, 'public/pwa-512x512.png')
    generate_icon(180, 'public/apple-touch-icon.png')
    generate_icon(64, 'public/favicon.png')

    # Also create a 32x32 favicon
    generate_icon(32, 'public/favicon.ico')

    print("\n✅ All icons generated successfully!")
    print("\nNote: These are placeholder icons. Consider replacing them with")
    print("professionally designed icons for production use.")

if __name__ == '__main__':
    try:
        main()
    except ImportError:
        print("Error: PIL (Pillow) library is required.")
        print("Install it with: pip3 install Pillow")
        exit(1)
    except Exception as e:
        print(f"Error: {e}")
        exit(1)

File: generate-icons.mjs
#!/usr/bin/env node
import fs from 'fs';
import { createCanvas } from 'canvas';

// Colors from the app
const bgColor = '#fffefb';
const primaryColor = '#00668c';
const accentColor = '#71c4ef';

function generateIcon(size, filepath) {
  const canvas = createCanvas(size, size);
  const ctx = canvas.getContext('2d');

  // Background
  ctx.fillStyle = bgColor;
  ctx.fillRect(0, 0, size, size);

  // Circle background
  const centerX = size / 2;
  const centerY = size / 2;
  const radius = size * 0.4;

  ctx.fillStyle = primaryColor;
  ctx.beginPath();
  ctx.arc(centerX, centerY, radius, 0, Math.PI * 2);
  ctx.fill();

  // Inner circle accent
  ctx.fillStyle = accentColor;
  ctx.beginPath();
  ctx.arc(centerX, centerY, radius * 0.6, 0, Math.PI * 2);
  ctx.fill();

  // Draw "V" letter
  ctx.fillStyle = '#ffffff';
  ctx.font = `bold ${size * 0.5}px Arial`;
  ctx.textAlign = 'center';
  ctx.textBaseline = 'middle';
  ctx.fillText('V', centerX, centerY);

  // Save to file
  const buffer = canvas.toBuffer('image/png');
  fs.writeFileSync(filepath, buffer);
  console.log(`Generated ${filepath}`);
}

// Generate icons
try {
  generateIcon(192, './public/pwa-192x192.png');
  generateIcon(512, './public/pwa-512x512.png');
  generateIcon(180, './public/apple-touch-icon.png');

  // Generate a simple favicon (16x16)
  const faviconSize = 32;
  const canvas = createCanvas(faviconSize, faviconSize);
  const ctx = canvas.getContext('2d');

  ctx.fillStyle = primaryColor;
  ctx.fillRect(0, 0, faviconSize, faviconSize);

  ctx.fillStyle = '#ffffff';
  ctx.font = `bold ${faviconSize * 0.7}px Arial`;
  ctx.textAlign = 'center';
  ctx.textBaseline = 'middle';
  ctx.fillText('V', faviconSize / 2, faviconSize / 2);

  const buffer = canvas.toBuffer('image/png');
  fs.writeFileSync('./public/favicon.png', buffer);
  console.log('Generated ./public/favicon.png');

  console.log('\nAll icons generated successfully!');
  console.log('You may want to replace these placeholder icons with professionally designed ones.');
} catch (error) {
  console.error('Error generating icons:', error);
  console.error('\nNote: This script requires the "canvas" package.');
  console.error('Install it with: npm install --save-dev canvas');
  process.exit(1);
}
